src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Options/CollectionRuleOptionsExtensions.cs

```diff
@@ -0,0 +1,270 @@
+            Assert.Null(ruleOptions.Trigger.Settings);
+        }
+
+        private static void VerifyTrigger(this CollectionRuleOptions ruleOptions, string triggerType)
+        {
+            Assert.NotNull(ruleOptions.Trigger);
+            Assert.Equal(triggerType, ruleOptions.Trigger.Type);
+        }
+
+        public static CollectDumpOptions VerifyCollectDumpAction(this CollectionRuleOptions ruleOptions, int actionIndex, DumpType expectedDumpType, string expectedEgress)
+        {
+            CollectDumpOptions collectDumpOptions = ruleOptions.VerifyAction<CollectDumpOptions>(
+                actionIndex, KnownCollectionRuleActions.CollectDump);
+
+            Assert.Equal(expectedDumpType, collectDumpOptions.Type);
+            Assert.Equal(expectedEgress, collectDumpOptions.Egress);
+
+            return collectDumpOptions;
+        }
+
+        public static CollectGCDumpOptions VerifyCollectGCDumpAction(this CollectionRuleOptions ruleOptions, int actionIndex, string expectedEgress)
+        {
+            CollectGCDumpOptions collectGCDumpOptions = ruleOptions.VerifyAction<CollectGCDumpOptions>(
+                actionIndex, KnownCollectionRuleActions.CollectGCDump);
+
+            Assert.Equal(expectedEgress, collectGCDumpOptions.Egress);
+
+            return collectGCDumpOptions;
+        }
+
+        public static CollectLogsOptions VerifyCollectLogsAction(this CollectionRuleOptions ruleOptions, int actionIndex, string expectedEgress)
+        {
+            CollectLogsOptions collectLogsOptions = ruleOptions.VerifyAction<CollectLogsOptions>(
+                actionIndex, KnownCollectionRuleActions.CollectLogs);
+
+            Assert.Equal(expectedEgress, collectLogsOptions.Egress);
+
+            return collectLogsOptions;
+        }
+
+        public static CollectTraceOptions VerifyCollectTraceAction(this CollectionRuleOptions ruleOptions, int actionIndex, TraceProfile expectedProfile, string expectedEgress)
+        {
+            CollectTraceOptions collectTraceOptions = ruleOptions.VerifyAction<CollectTraceOptions>(
+                actionIndex, KnownCollectionRuleActions.CollectTrace);
+
+            Assert.Equal(expectedProfile, collectTraceOptions.Profile);
+            Assert.Equal(expectedEgress, collectTraceOptions.Egress);
+
+            return collectTraceOptions;
+        }
+
+        public static CollectTraceOptions VerifyCollectTraceAction(this CollectionRuleOptions ruleOptions, int actionIndex, IEnumerable<EventPipeProvider> providers, string expectedEgress)
+        {
+            CollectTraceOptions collectTraceOptions = ruleOptions.VerifyAction<CollectTraceOptions>(
+                actionIndex, KnownCollectionRuleActions.CollectTrace);
+
+            Assert.Equal(expectedEgress, collectTraceOptions.Egress);
+            Assert.NotNull(collectTraceOptions.Providers);
+            Assert.Equal(providers.Count(), collectTraceOptions.Providers.Count);
+
+            int index = 0;
```

Comment:
It looks like this isn't being incremented (and I'm assuming it should be...?)

Reply:
Fixed